parsl/providers/azure/azure.py

```diff
@@ -0,0 +1,489 @@
+
+logger = logging.getLogger(__name__)
+
+try:
+    from azure.common.credentials import ServicePrincipalCredentials
+    from azure.mgmt.resource import ResourceManagementClient
+    from azure.mgmt.network import NetworkManagementClient
+    from azure.mgmt.compute import ComputeManagementClient
+    from azure.mgmt.compute.models import DiskCreateOption
+
+    _api_enabled = True
+
+except ImportError:
+    _api_enabled = False
+else:
+    _api_enabled = True
+
+translate_table = {
+    'VM pending': 'PENDING',
+    'VM running': 'RUNNING',
+    'VM deallocated': 'COMPLETED',
+    'VM stopping': 'COMPLETED',  # We shouldn't really see this state
+    'VM stopped': 'COMPLETED',  # We shouldn't really see this state
+}
+
+
+class AzureProvider(ExecutionProvider, RepresentationMixin):
+    """
+    A Provider for using Microsoft Azure Resources
+
+    One of 2 methods are required to authenticate: keyfile, or environment
+    variables. If  keyfile is not set, the following environment
+    variables must be set: `AZURE_CLIENT_ID` (the access key for
+    your azure account),
+    `AZURE_CLIENT_SECRET` (the secret key for your azure account), the
+    `AZURE_TENANT_ID` (the session key for your azure account), and
+    AZURE_SUBSCRIPTION_ID.
+
+    Parameters
+    ----------
+    vm_reference : dict
+        Dictionary describing the parameters of the Azure VM.
+
+        Required structure:
+        {
+          'publisher': VM OS publisher
+          'offer': VM OS offer
+          'sku': VM OS SKU
+          'version': VM OS version
+          'vm_size': VM Size, analogous to instance type for AWS
+          'disk_size_gb': (int) size of VM disk in gb
+          "admin_username": (str) admin username of VM instances,
+          "password": (str) admin password for VM instances
+        }
+    worker_init : str
+        String to append to the Userdata script executed in the cloudinit phase of
+        instance initialization.
+    walltime : str
+        Walltime requested per block in HH:MM:SS.
+    key_file : str
+        Path to json file that contains 'Azure keys'
+    nodes_per_block : int
+        This is always 1 for Azure. Nodes to provision per block.
+    nodes_per_block : int
+        Nodes to provision per block. Default is 1.
+    init_blocks : int
+        Number of blocks to provision at the start of the run. Default is 1.
+    min_blocks : int
+        Minimum number of blocks to maintain. Default is 0.
+    max_blocks : int
+        Maximum number of blocks to maintain. Default is 10.
+    region : str
+        Azure region to launch machines. Default is 'westus'.
+    key_name : str
+        Name of the Azure private key (.pem file) that is usually generated on the console
+        to allow SSH access to the Azure instances. This is mostly used for debugging.
+    walltime : str
+        Walltime requested per block in HH:MM:SS. This option is not currently honored by this provider.
```

Comment:
I think it would be better to remove this parameter rather than document it as being ignored.

Reply:
I mirrored this from the AWS provider. Is this planned to be a feature in the future? And if so, is it better to leave it in so that the constructor will be invoked the same way in future releases?

Reply:
Cloud providers don't generally work with walltime constraints like batch systems. Let's remove this argument.

Reply:
even if we do support it in future, my feeling is that compatibility is better by removing it now and adding it back in with a sensible default when there is meaning.

the present layout encourages people to specify walltimes using the wrong syntax, which will break if it is ever implemented; to puzzle over why it is not respected; and in potential-future to puzzle over why their jobs are suddenly terminated after the wrong walltime after upgrading parsl.

as long as the keyword has a default value if ever added back in, then calls to the initializer without `walltime` will still behave correctly in that potential-future version.

Reply:
Got it - sounds good.